tests/test_mockgun.py

```diff
@@ -269,6 +272,91 @@ def test_find_by_sub_entity_field(self):
+
+        # Create a project for nested sub entity field
+        self._project1 = self._mockgun.create("Project", {"name": "project1", "users": [self._user1]})
+
+        # Create pipeline configurations that are assigned none, one or two users.
+        self._mockgun.create(
+            "PipelineConfiguration",
+            {"code": "with_user1", "users": [self._user1], "project": self._project1}
+        )
+
+    def test_field(self):
+        item = self._mockgun.find_one("HumanUser", [
+            ["login", "is", "user1"]
+        ], ["firstname"])
+        self.assertIn("firstname",item)
+        self.assertEqual("firstname1", item["firstname"])
+
+    def test_find_field_multi_entity(self):
+        item = self._mockgun.find_one("PipelineConfiguration", [
+            ["project.Project.users.HumanUser.login", "is", "user1"]
+        ], ["project.Project.name"])
+        self.assertIn("project.Project.name", item)
+        self.assertEqual("project1", item["project.Project.name"])
+
+
+class TestFindOrder(TestBaseWithExceptionTests):
+    """
+    Ensure that using the 'order' argument in the find method work.
+    """
+    def setUp(self):
+        self._mockgun = Mockgun("https://test.shotgunstudio.com", login="user", password="1234")
+
+        # Create two users to assign to the pipeline configurations.
+        self._user1 = self._mockgun.create("HumanUser", {"login": "user1", "firstname": "firstname1"})
+        self._user2 = self._mockgun.create("HumanUser", {"login": "user2", "firstname": "firstname2"})
+
+        # Create two PipelineConfiguration entities so we can test multi-entity sorting.
+        self._pipeline_configutation_1 = self._mockgun.create("PipelineConfiguration", {"code": "PipelineConfiguration1", "users": [self._user2]})
+        self._pipeline_configutation_2 = self._mockgun.create("PipelineConfiguration", {"code": "PipelineConfiguration2", "users": [self._user1]})
+
+    def test_find_order(self):
+        # Test ascending order
+        item = self._mockgun.find_one("HumanUser", [], order=[{'field_name': 'id', 'direction': 'asc'}])
+        self.assertEqual(self._user1['id'], item['id'])
+
+        # Test descending order
+        item = self._mockgun.find_one("HumanUser", [], order=[{'field_name': 'id', 'direction': 'desc'}])
+        self.assertEqual(self._user2['id'], item['id'])
+
+    def test_find_order_multi_entity(self):
+        """Ensure we are able to sort data using a multi-entity field."""
+        # Test ascending order
+        item = self._mockgun.find_one("PipelineConfiguration", [], order=[{'field_name': 'users.HumanUser.login', 'direction': 'asc'}])
+        self.assertEqual(self._pipeline_configutation_2['id'], item['id'])
+
+        # Test descending order
+        item = self._mockgun.find_one("PipelineConfiguration", [], order=[{'field_name': 'users.HumanUser.login', 'direction': 'desc'}])
```

Comment:
We are using a lot of sorting on nested entity field on our side.
For example in production we have a custom entity named Episode that contain Sequences.
Here is the code I use to get all of it's sequence in the correct order.
```
filters = [
["sg_sequence.Sequence.sg_episode", "is", {'type': self.type, 'id': self.id}]
]
order = [
    {'field_name': 'sg_sequence.Sequence.sg_sequence_order', 'direction': 'asc'},
    {'field_name': 'sg_cut_order', 'direction': 'asc'},
]
```
However I just realised that the sg_episode and sg_sequence are 'entity' fields and the PipelineConfiguration.users is an 'multi_entity' fields. You might be right it may be not possible to sort using 'multi_entity' fields. 

I was using the 'PipelineConfigure.users' field to have a simple test case without the need to create custom fields, however I might have to rename 'test_find_order_multi_entity' to 'test_find_order_nested_entity' and design a better test case that exclusively use 'entity' fields instead of 'multi_entity' fields. I'll think of something and update the pull request accordingly!